rocker-compiler/src/main/java/com/fizzed/rocker/compiler/JavaGenerator.java

```diff
@@ -1053,6 +1053,80 @@ else if (unit instanceof SwitchDefaultBlockEnd) {
                         .append(blockEnd.pop())
                         .append(" // default end ").append(sourceRef(unit)).append(CRLF);
             }
+
+            else if(unit instanceof SwitchExpressionBlock){
+
+                SwitchExpressionBlock block = (SwitchExpressionBlock) unit;
+
+                // break support via try and catch mechanism (works across lambdas!)
+                tab(w, depth+indent)
+                        .append("try {").append(CRLF);
+
+                depth++;
+
+                tab(w, depth+indent)
+                        .append("switch ")
+                        .append(block.getExpression())
+                        .append(" {").append(CRLF);
+
+                blockEnd.push("}");
+
+                depth++;
+            } else if (unit instanceof SwitchExpressionBlockEnd) {
+                depth--;
+
+                tab(w, depth+indent)
+                        .append(blockEnd.pop())
+                        .append(" // switch end ").append(sourceRef(unit)).append(CRLF);
+
+                depth--;
+
+                // break support via try and catch mechanism (works across lambdas!)
+                tab(w, depth+indent)
+                        .append("} catch (").append(BreakException.class.getCanonicalName()).append(" e) {") .append(CRLF);
+
+                tab(w, depth+indent+1)
+                        .append("// support for breaking switch statements").append(CRLF);
+
+                tab(w, depth+indent)
+                        .append("}").append(CRLF);
+
+            } else if (unit instanceof SwitchCaseExpressionBlock) {
+
+                SwitchCaseExpressionBlock block = (SwitchCaseExpressionBlock) unit;
+
+                tab(w, depth+indent)
+                        .append("case ")
+                        .append(block.getExpression())
+                        .append("-> ")
```

Comment:
Let's add a single space char before the -> so that the generated java code matches the style used for other parts such as "default ->".  Right now it generates something like "case "a"-> "

Reply:
done